core/block_validator.go

```diff
@@ -135,13 +135,20 @@ func (v *BlockValidator) ValidateState(block *types.Block, statedb *state.StateD
 			return nil
 		},
 	}
-	if skipHeavyVerify {
+	if statedb.IsPipeCommit() {
 		validateFuns = append(validateFuns, func() error {
 			if err := statedb.WaitPipeVerification(); err != nil {
 				return err
 			}
+			statedb.CorrectAccountsRoot()
+
+			diffLayer := &types.DiffLayer{}
```

Comment:
I can understand that you want to cache the difflayer with the correct storage root,  but difflayer is used for diffsync, diff sync does not need storage root.